### PR TITLE
Implement blocks pool

### PR DESCRIPTION
**Pull request by ssd04**

### CODE REVIEW

File: process/pruningStorer.go [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+
+	err := ps.putInPersister([]byte(metaCheckpointKey), metaRound)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ps *pruningStorer) GetCheckpoint() (uint64, error) {
```
Comment (ssd04) [inline process/pruningStorer.go:245]: these checkpoint functionalities will be handled separately in next PR

File: process/blocksPool.go [hunk 2]
```diff
@@ -2,44 +2,183 @@ package process
...
+		}
+
+		bp.indexesMap[shardID] = index
+
+		return nil
+	}
+
+	metaIndex := bp.indexesMap[core.MetachainShardId]
+
+	if !bp.shouldPutBlockData(currentIndex, metaIndex) {
```
[the remark this answers is not in the document]
Comment (ssd04) [reply process/blocksPool.go:130]: in the next PR we can have this functionality based on last checkpoint data

File: process/blocksPool.go [hunk 3]
```diff
@@ -2,44 +2,183 @@ package process
...
+// Get will get value from storer
+func (bp *blocksPool) Get(key []byte) ([]byte, error) {
+	return bp.storer.Get(key)
+}
+
+func (bp *blocksPool) UpdateMetaState(index uint64) {
+	bp.mutMap.Lock()
+	defer bp.mutMap.Unlock()
+
+	bp.indexesMap[core.MetachainShardId] = index
```
[the remark this answers is not in the document]
Comment (ssd04) [reply process/blocksPool.go:79]: right, refactored mutex usage

File: factory/dataProcessorFactory.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+		Capacity:    cfg.OutportBlocksStorage.Cache.Capacity,
+	}
+
+	cacher, err := storageUnit.NewCache(cacheConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	switch dbMode {
+	case FullPersisterDBMode:
```
[the remark this answers is not in the document]
Comment (ssd04) [reply factory/dataProcessorFactory.go:56]: `FullPersister` mode is when the block will be saved also to storage every time

File: cmd/connector/config/config.toml [hunk 5]
```diff
@@ -1,20 +1,52 @@
...
     # This flag specifies if we should drop messages if there is no connection to the host
-    drop_messages_if_no_connection = false
-    version = 1
+    DropMessagesIfNoConnection = false
+
+    # Version specifies payload version (default = 1)
+    Version = 1
+
+[DataPool]
+    NumberOfShards = 3
```
[the remark this answers is not in the document]
Comment (ssd04) [reply cmd/connector/config/config.toml:32]: removed number of shards from here

**Review by cristure — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**